temba/utils/s3/views.py

```diff
@@ -0,0 +1,23 @@
+from django.http import FileResponse, Http404
+from django.views import View
+from django.views.decorators.csrf import csrf_exempt
+
+from temba.utils.s3 import private_file_storage
+
+
+class PrivateFileCallbackView(View):
+    """
+    When we use the AWS S3 to send attachments, we send relative URLs which user later GETs to get the
+    file content.
+    """
+
+    @csrf_exempt
+    def dispatch(self, *args, **kwargs):
+        return super().dispatch(*args, **kwargs)
+
+    @staticmethod
+    def get(request, *args, **kwargs):
+        try:
+            return FileResponse(private_file_storage.open(kwargs["file_path"]))
```

Comment:
Nice job @ihor-palii. Will we need to make any migration file to update the existing media fields with this endpoint? I mean, we need to be concerned about it, right, since we are going to make the entire bucket private?

Reply:
I think we can create a migration that can update the URLs. Possible issues when migrating URLs in flow definitions so we need to be careful with it. Another option that I see is to create new private storage e.g. 'ccl-dev-private' and replace the existing configuration. In that case, all the existing URLs inside the flow definitions still will be active. And the new ones will use private storage. What do you think which option is better?

Reply:
I agree that the first option is really dangerous, it may mess things up. The other option that you suggested seems to be a good one, I'll need to justify to Vanta that the current bucket will not be used in production anymore but I think we can go with this solution of having another private bucket for the new attached files. We just need to make sure that the old one will be functioning as normal as before for old files.